apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+      return;
+    }
+
+    this.elemList = list;
+    window.removeEventListener('keydown', this);
+    window.addEventListener('keydown', this);
+  };
+
+
+  PinAppNavigation.prototype = {
+
+    selector: '',
+    elements: null,
+    preventDef: false,
+    selectedElemIndex: 0,
+    elemList: null,
+    prevVerticalKey: null,
+
+    middleElem: false,
+
+
+    /**
+     * This function reset all pinns to default values.
+     * @return {[nothing]} [nothing]
+     */
+    reset: function reset() {
+      MoreAppsNavigation.clearNavigationEvents();
+      this.elements.sort(function(elem1, elem2) {
+        if (!elem1.dataset.index || !elem2.dataset.index) {
+          return 1;
+        }
+        return elem1.dataset.index - elem2.dataset.index;
+      });
+
+
+      var selected = this.elemList.querySelector('.selected');
+
+      if (selected) {
+        selected.classList.remove('selected')
+      }
+
+      this.selectedElemIndex = 0;
+
+      for (var elemIndex in this.elements) {
+        this.elemList.appendChild(this.elements[elemIndex]);
+      }
+
+      this.elements[this.selectedElemIndex].classList.add('selected');
+      this.elements[this.selectedElemIndex].classList.add('middle');
+      this.elemList.removeAttribute('style');
+      this.middleElem = true;
+
+      var toRemoved = this.elemList.querySelector('.removed');
+      var toRemoveOutFocus = this.elemList.querySelector('.out-focus');
+
+      if (toRemoved) {
+        this.elemList.removeChild(toRemoved);
+      }
+
+      if (toRemoveOutFocus) {
+        toRemoveOutFocus.classList.remove('out-focus');
+      }
+
+      window.removeEventListener('keydown', this);
+      window.addEventListener('keydown', this);
+    },
+
+
+    /**
+     * This function refresh elements by selector
+     * @return {[nothins]} [nothing]
+     */
+    refresh: function refresh() {
+      if (!this.selector) {
+        console.log('Can not work without selector');
+        return;
+      }
+
+      this.elements = Array.prototype.slice.call(document.querySelectorAll(this.selector));
+
+      if (!this.elements && !this.elements.length) {
+        console.log('Can not find any elements by this selector');
+        return;
+      }
+
+      this.elements[this.selectedElemIndex].classList.add('selected');
+        this.elements[this.selectedElemIndex].classList.add('middle');
+        this.middleElem = true;
+    },
+
+
+    /**
+     * Handle event keydown
+     * @param  {[Object]} e [default object]
+     * @return {[nothing]}   [nothing]
+     */
+    handleEvent: function(e) {
+
+      function cycleElements(index) {
+
+        var removed = this.elements.splice(-index, index);
+
+        for (var i = 0; i < removed.length; i++) {
+
+          this.elemList.insertBefore(removed[i], this.elements[0 + i]);
+          this.elements.splice(0 + i, 0, removed[i]);
+
+        }
+
+      };
+
+      if (this.preventDef) {
+        e.preventDefault();
```

Comment:
You really can't prevent default until you've checked the keycode and know you're going to handle the key. This will break the volume keys, for example.